bugsnag-android-core/src/main/java/com/bugsnag/android/DeliveryDelegate.java

```diff
@@ -43,12 +43,32 @@ void deliver(@NonNull Event event) {
         }
 
         if (event.isUnhandled()) {
-            cacheEvent(event, true);
+            // should only send unhandled errors if they don't terminate the process or
+            // if they can't be reported on next launch. ANRs don't terminate the process
+            // directly and stackoverflow can't be persisted on disk
+            boolean sendInSameSession = isAnr(event) || isStackoverflow(event);
+            cacheEvent(event, sendInSameSession);
         } else {
             deliverPayloadAsync(event, eventPayload);
         }
     }
 
+    boolean isAnr(@NonNull Event event) {
+        List<Error> errors = event.getErrors();
+        String errorClass = null;
+
+        if (!errors.isEmpty()) {
+            Error error = errors.get(0);
+            errorClass = error.getErrorClass();
+        }
+        return "ANR".equals(errorClass);
+    }
+
+    boolean isStackoverflow(@NonNull Event event) {
+        Throwable err = event.getOriginalError();
+        return (err != null) && StackOverflowError.class.equals(err.getClass());
+    }
```

Comment:
These feel like useful methods that could be on `Event` - can they be made protected methods in there?

Reply:
Moved method to `Event`